tardis/model/tests/test_csvy_model.py

```diff
@@ -34,24 +45,298 @@ def test_compare_models(filename):
+    reference_input_abundance = pd.DataFrame(
+        [[0.0, 0.33, 0.3, 0.5, 0.4, 0.2], [0.98, 0.64, 0.6, 0.4, 0.55, 0.79]],
+        index=abundance_index,
+    )
+ 
+    arrays = [[28], [56]]
+    isotope_index = pd.MultiIndex.from_arrays(
+        arrays, names=["atomic_number", "mass_number"]
+    )
+    reference_input_isotopes = pd.DataFrame(
+        [[0.02, 0.03, 0.1, 0.1, 0.05, 0.01]], 
+        columns=np.arange(6), index=isotope_index
+    )
+    return reference_input_abundance,reference_input_isotopes
+
+def test_read_csvy_abundances(csvy_model_test_abundances,
+                            reference_input_dataframes):
+    """Test if model reads abundances and isotope abundances 
+       and constructs dataframes correctly before applying decay"""
+    reference_input_abundance,reference_input_isotopes = reference_input_dataframes
+    
+    model_abundance_shape = csvy_model_test_abundances.raw_abundance.shape
+    reference_input_shape = reference_input_abundance.shape
+    assert model_abundance_shape == reference_input_shape
+    npt.assert_array_almost_equal(
+      reference_input_abundance.to_numpy(), 
+      csvy_model_test_abundances.raw_abundance.to_numpy()
+    )
+
+    model_isotopes_shape = csvy_model_test_abundances.raw_isotope_abundance.shape
+    reference_input_isotopes_shape = reference_input_isotopes.shape
+    assert model_isotopes_shape == reference_input_isotopes_shape
+    npt.assert_array_almost_equal(
+        reference_input_isotopes.to_numpy(),
+        csvy_model_test_abundances.raw_isotope_abundance.to_numpy(),
+    )
+
+
+@pytest.fixture(scope="function")
+def reference_decayed_abundance():
+    """Returns reference abundances dataframe for `test_csvy_model_decay`
+    Constructed from `csvy_model_test_abundances.yml` """
+    decay_index = pd.Index([1, 2, 26, 27, 28], name="atomic_number")
+    reference_decayed_abundance = pd.DataFrame(
+        [
+            [0.0, 0.33, 0.3, 0.5, 0.4, 0.2],
+            [0.98, 0.64, 0.6, 0.4, 0.55, 0.79],
+            [
+                0.00013977843354947162,
+                0.00020966765032420787,
+                0.0006988921677473642,
+                0.0006988921677473642,
+                0.0003494460838736821,
+                6.988921677473581e-05,
+            ],
+            [
+                0.007188928223953217,
+                0.010783392335929825,
+                0.035944641119766085,
+                0.035944641119766085,
+                0.017972320559883043,
+                0.0035944641119766084,
+            ],
+            [
+                0.012671293342497312,
+                0.019006940013745966,
+                0.06335646671248656,
+                0.06335646671248656,
+                0.03167823335624328,
+                0.006335646671248656,
+            ],
+        ],
+        index=decay_index,
+    )
+    return reference_decayed_abundance
+
+def test_csvy_model_decay(csvy_model_test_abundances,
```

Comment:
I think this function has been duplicated?  Why did this happen and which one is correct?